### ISSUE TITLE
Rock Pi S CPU settings

### ISSUE THREAD

**Issue by BigMuscle85** [radxa-repo/rbuild#41]
When building the image for Rock Pi S, it does not provide full CPU functionality:
- CPU runs on 1 GHz maximally although it should be 1.3GHz
```
rock@rock-pi-s:~$ cat /sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies
408000 600000 816000 1008000
```

- CPU temperature not available and thermal zones in /sys/class/thermal/ are missing. This is related to the following errors reported in  the kernel log. The problem here is that kernel 6.x requires 'trips' to be defined for all thermal zones else it won't load them at all.
> [   14.912964] thermal_sys: Failed to find 'trips' node
> [   14.913472] thermal_sys: Failed to find trip points for tsadc id=0
> [   14.914068] rockchip-thermal ff1f0000.tsadc: failed to register sensor 0: -22
> [   14.914737] rockchip-thermal ff1f0000.tsadc: failed to register sensor[0] : error = -22

https://github.com/radxa-repo/bsp/blob/main/linux/.stable-6.1/0023-armbian-rockpi-s/board-rockpis-0007-arm64-dts-rockchip-add-cpu-s-thermal-config-for-rk33.patch#L56

I've created an additional overlay to make this work:

```
/dts-v1/;
/plugin/;

/ {
    compatible = "rockchip,rk3308";

    fragment@0 {
                target = <&thermal_zones>;
                __overlay__ {
                        logic-thermal {
                                thermal-sensors = <&tsadc 0>;

                                trips {
                                        temp_crit: temp_crit {
                                               temperature = <90000>;
                                               hysteresis = <2000>;
                                               type = "critical";
                                        };
                                };
                        };
                };
    };
};
```

**Comment by RadxaYuntian** [comment]
> CPU runs on 1 GHz maximally although it should be 1.3GHz

Only RK3308B-S variant could support this speed. If your ROCK Pi S is indeed using that SoC, you can use [this overlay](https://github.com/radxa/overlays/blob/main/arch/arm64/boot/dts/rockchip/overlays/rk3308-bs-opp-1296.dts).

Prebuilt dtbo are available in GitHub Workflow's artifact section.

> CPU temperature not available and thermal zones in /sys/class/thermal/ are missing. This is related to the following errors reported in the kernel log. The problem here is that kernel 6.x requires 'trips' to be defined for all thermal zones else it won't load them at all.

Thanks for spotting this. I'll add this to our source tomorrow.

**Comment by BigMuscle85** [comment]
Thank you for the information. I didn't know that B-S variant is needed for 1.3 GHz as doc here https://wiki.radxa.com/RockpiS does not mention it. 
There's RK3308B written on my chip (board is V1.3) and running Armbian with 1.3 GHz overlay runs correctly even under heavy load. But on Radxa image, rsetup does not offer the overlay to be enabled at all, so I guess I need to enable it manually?

**Comment by RadxaYuntian** [comment]
We have ceased RK3308B-S production due to many incompatibilities with existing RK3308 design, so I very much doubt yours is indeed 3308B-S. Having RK3308B written on it does not mean it is the B-S variant.

RK3308B-S either says RK3308B-S directly on the chip, or says RK3308B, but the second line will ends with `S` before the production week begins.

Given below chip as an example, the 2nd line is SBAKW05003. Since it is ending with `3` it is not RK3308B-S.

![image](https://github.com/radxa-repo/rbuild/assets/95260730/773b7341-b22f-46dc-8a1d-5cd2ecff38f8)

**Comment by BigMuscle85** [comment]
You are right. Our chips also ends with `3` and not `S`. So it's definitely not B-S version. It is just a bit confusing because the official documentation on Radxa website talks about 1.3 GHz regardless the chip version and also the vendor sells all Rock Pi S as 1.3 GHz CPU. So the specified frequency is not correct and all boards now support only 1 GHz?

**Comment by RadxaYuntian** [comment]
I'd think so. If you can please send me the links of 1.3 GHz so I could get them updated.

**Comment by BigMuscle85** [comment]
It is here:
https://wiki.radxa.com/RockpiS/getting_started
https://radxa.com/products/rocks/pis
https://radxa.com/products/rocks/pis#techspec
+ Product Brief PDF downloadable on that website

This doc says that revisions 1.3 since 2022 (ours are V1.3, week 2311) are B-S - https://wiki.radxa.com/RockpiS/hardware/revisions

**Comment by RadxaYuntian** [comment]
Thanks. @melooooooooooo please take a look at this.

**Comment by melooooooooooo** [comment]
Thank you for your support of Radxa.
The information you mentioned has been updated.
